### PR TITLE
core(theme) Theme inheritance PoC 

### PR DESCRIPTION
**Pull request by filrak**
Once there is a file in `/app/{dir}` it overrides `nuxt-module/theme/{dir}`

### CODE REVIEW

File: packages/core/theme-module/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+const log = {
+  info: (message) => consola.info(chalk.bold('VSF'), message),
+  success: (message) => consola.success(chalk.bold('VSF'), message),
+  warning: (message) => consola.warning(chalk.bold('VSF'), message),
+  error: (message) => consola.error(chalk.bold('VSF'), message)
+}
+
+const getAllFiles = function(dirPath) {
+  files = fs.readdirSync(dirPath)
```
Comment (defudef) [inline packages/core/theme-module/index.js]: missing const

File: packages/core/theme-module/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ 
+  arrayOfFiles = []
```
Comment (defudef) [inline packages/core/theme-module/index.js]: missing let

File: packages/core/theme-module/index.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+ 
+  files.forEach(function(file) {
```
Comment (defudef) [inline packages/core/theme-module/index.js]: Just use map:

```JS
files.map(file => {
   if (fs.statSync(`${dirPath}/${file}`).isDirectory()) {
     return ...getAllFiles(dirPath + '/' + file, arrayOfFiles)
   } else {
      return`${dirPath}/${file}`.split(__dirname + '/').pop()
})
```

File: packages/core/theme-module/index.js (hunk 1 above)
Comment (andrzejewsky) [reply packages/core/theme-module/index.js]: he wants here a global one : )

File: packages/core/theme-module/index.js (hunk 3 above)
Comment (andrzejewsky) [reply packages/core/theme-module/index.js]: I would say reduce...  but... it's just a proposition:
```
return files.reduce((prev, curr) => {
  if (fs.statSync(dirPath + "/" + curr).isDirectory()) {
    return getAllFiles(dirPath + "/" + curr, prev)
  }

  const newFiles = (dirPath + "/" + curr).split(__dirname + '/').pop()

  return [...prev, ...newFiles]
}, [])
```

**Review by andrzejewsky — APPROVED** [review]
remove error log

**Pull request merged**